File: app/src/main/java/com/mtime/player/ReceiverGroupManager.java
package com.mtime.player;

import android.content.Context;

import com.kk.taurus.playerbase.receiver.ReceiverGroup;
import com.mtime.common.utils.PrefsManager;
import com.mtime.player.receivers.DataReceiver;
import com.mtime.player.receivers.LogReceiver;
import com.mtime.player.receivers.NewControllerCover;
import com.mtime.player.receivers.ErrorCover;
import com.mtime.player.receivers.LoadingCover;
import com.mtime.player.receivers.PauseAdCover;
import com.mtime.player.receivers.PlayerGestureCover;
import com.mtime.player.receivers.UserGuideCover;
import com.mtime.player.receivers.VideoDefinitionCover;

/**
 * Created by JiaJunHui on 2018/6/14.
 */
public class ReceiverGroupManager {

    public static final String FIRST_ENTRY_FULL_SCREEN = "first_entry_full_screen";

    public static ReceiverGroup getBaseReceiverGroup(Context context) {
        ReceiverGroup receiverGroup = new ReceiverGroup();
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_DATA_RECEIVER, new DataReceiver(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_LOG_RECEIVER, new LogReceiver(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_LOADING_COVER, new LoadingCover(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_CONTROLLER_COVER, new NewControllerCover(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_ERROR_COVER, new ErrorCover(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_PAUSE_AD_COVER, new PauseAdCover(context));
        return receiverGroup;
    }

    public static ReceiverGroup getStandardReceiverGroup(Context context) {
        ReceiverGroup receiverGroup = getBaseReceiverGroup(context);
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_DEFINITION_COVER, new VideoDefinitionCover(context));
        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_GESTURE_COVER, new PlayerGestureCover(context));
        if (isNeedUserGuideCover(context))
            receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_USER_GUIDE_COVER, new UserGuideCover(context));
        return receiverGroup;
    }

    public static boolean isNeedUserGuideCover(Context context) {
        return !PrefsManager.get(context).getBoolean(FIRST_ENTRY_FULL_SCREEN);
    }

    public static void updateUserGuideIKnow(Context context) {
        PrefsManager.get(context).putBoolean(FIRST_ENTRY_FULL_SCREEN, true);
    }

}

File: player/src/main/java/com/kotlin/android/player/ReceiverGroupManager.kt
package com.kotlin.android.player

import android.content.Context
import com.kk.taurus.playerbase.receiver.ReceiverGroup
import com.kk.taurus.playerbase.touch.OnTouchGestureListener
import com.kotlin.android.core.ext.getSpValue
import com.kotlin.android.core.ext.putSpValue
import com.kotlin.android.player.receivers.*

/**
 * @author ZhouSuQiang
 * @email zhousuqiang@126.com
 * @date 2020/7/15
 *
 * 播放器UI组件拼装
 */
object ReceiverGroupManager {
    const val FIRST_ENTRY_FULL_SCREEN = "first_entry_full_screen"
    /**
     * 获取极简UI组件
     */
    fun getLessReceiverGroup(context: Context, listener: OnTouchGestureListener): ReceiverGroup {
        return ReceiverGroup().apply {
            addReceiver(DataInter.ReceiverKey.KEY_TOUCH_GESTURE_COVER, TouchGestureListenerReceiver(context, listener))
            addReceiver(DataInter.ReceiverKey.KEY_LOADING_COVER, LoadingCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_ERROR_COVER, ErrorCover(context))
        }
    }

    fun getNormalReceiverGroup(context: Context):ReceiverGroup{
        return ReceiverGroup().apply {
            addReceiver(DataInter.ReceiverKey.KEY_LOADING_COVER, LoadingCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_ERROR_COVER, ErrorCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_CONTROLLER_COVER, NewControllerCover(context))
        }
    }

    /**
     * 获取基础视频控制控件
     */
    fun getBaseReceiverGroup(context: Context):ReceiverGroup{
        return ReceiverGroup().apply {
            addReceiver(DataInter.ReceiverKey.KEY_DATA_RECEIVER, DataReceiver(context))
//        日志数据上报
//        receiverGroup.addReceiver(DataInter.ReceiverKey.KEY_LOG_RECEIVER, LogReceiver(context))
            addReceiver(DataInter.ReceiverKey.KEY_LOADING_COVER, LoadingCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_CONTROLLER_COVER, NewControllerCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_ERROR_COVER, ErrorCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_PAUSE_AD_COVER, PauseAdCover(context))
        }

    }

    /**
     * 获取标准视频控件
     */
    fun getStandardReceiverGroup(context: Context):ReceiverGroup{
        return getBaseReceiverGroup(context).apply {
            addReceiver(DataInter.ReceiverKey.KEY_DEFINITION_COVER, VideoDefinitionCover(context))
            addReceiver(DataInter.ReceiverKey.KEY_GESTURE_COVER, PlayerGestureCover(context))
            if (isNeeduserGuideCover()) addReceiver(DataInter.ReceiverKey.KEY_USER_GUIDE_COVER, UserGuideCover(context))
        }
    }

    fun getStandardWithoutGestureReceiverGroup(context: Context):ReceiverGroup{
        return getBaseReceiverGroup(context).apply {
            addReceiver(DataInter.ReceiverKey.KEY_DEFINITION_COVER, VideoDefinitionCover(context))
            if (isNeeduserGuideCover()) addReceiver(DataInter.ReceiverKey.KEY_USER_GUIDE_COVER, UserGuideCover(context))
        }
    }

    /**
     * 是否需要新手引导
     */
    fun isNeeduserGuideCover() = getSpValue(FIRST_ENTRY_FULL_SCREEN, true).not()

    fun updateUserGuideIKnow() {
        putSpValue(FIRST_ENTRY_FULL_SCREEN, true)
    }
}